src/auth.nim

```diff
@@ -227,20 +198,45 @@ proc initAccountPool*(cfg: Config; path: string) =
   elif fileExists(path):
     log "Parsing JSON guest accounts file: ", path
     accountPool = parseGuestAccounts(path)
-  elif guestAccountsUrl == "":
-    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests."
+  elif not cfg.guestAccountsUsePool:
+    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests. Alternatively, configure the guest account pool in nitter.conf"
     quit 1
 
-  let accountsPrePurge = accountPool.len
   accountPool.keepItIf(not it.hasExpired)
 
   log "Successfully added ", accountPool.len, " valid accounts."
 
+proc updateAccountPool*(cfg: Config) {.async.} =
+  if not cfg.guestAccountsUsePool:
+    return
+
+  while true:
+    if accountPool.len == 0:
+      let pool = HttpPool()
+
+      log "fetching more accounts from service"
+      pool.use(newHttpHeaders()):
+        let resp = await c.get("$1?id=$2&auth=$3" % [cfg.guestAccountsPoolUrl, cfg.guestAccountsPoolId, cfg.guestAccountsPoolAuth])
```

Comment:
If you make `cfg.guestAccountsPoolUrl` a Uri object from std/uri and parse it using `parseUri` in config.nim, you can do:
```suggestion
        let resp = await c.get($(cfg.guestAccountsPoolUrl ? {"id": cfg.guestAccountsPoolId, "auth": cfg.guestAccountsPoolAuth}))
```
It's not a big improvement by any means, but it's more consistent with how URLs are handled in the rest of Nitter's code. The `$(...)` part is only necessary because `safeFetch` from media.nim as mentioned above takes a string instead of a Uri, I'll move and fix the function after this PR gets merged.